### PR TITLE
Standardizes JS formatting with PrettierX

### PR DESCRIPTION
**Pull request by itsmeow**
## About The Pull Request

RIP Everyone's Git Blames (RIPEGB)

Ports:
- https://github.com/tgstation/tgstation/pull/67935

Touches literally every TGUI file, please speedmerge and cope with the conflicts (the conflict resolution is literally to hit "Use current" and then save the file again.)

This doesn't affect the actual logic of anything, so it's basically a free merge. If it affects the logic something is very very very very wrong. But that would be nonsense. It just reformats the code programatically.

### HOW TO PORT FOR DOWNSTREAMS

Cherry-pick `e1f4cf264bff4f1df41dce764866224011fffbc5` and `dbdf72ec734910832841352193a40f531de68837`.

In the repo directory run:
`cd tgui`
`yarn prettierx --write packages`

## Why It's Good For The Game

Allows for editing TGUI files without having to spam "Save without formatting" and provides a consistent code style for all TGUI files, also eliminates bugs with ESLint getting angry about how you formatted your file.

This will also fix style inconsistencies with modern TG code allowing for simpler ports - as well as easier rapid iteration on UIs since you don't have to cope with (or even think about) the formatting

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image

</details>

## Changelog
:cl: itsmeow, jlsnow301
code: Standardized TGUI code formatting using PrettierX. Developers: Install the Prettier extension for VSCode.
/:cl:

### CODE REVIEW

File: tgui/packages/common/fp.js [hunk 1]
```diff
@@ -9,19 +9,20 @@
...
  * value of the previous.
  */
-export const flow = (...funcs) => (input, ...rest) => {
-  let output = input;
-  for (let func of funcs) {
-    // Recurse into the array of functions
-    if (Array.isArray(func)) {
-      output = flow(...func)(output, ...rest);
+export const flow =
+  (...funcs) =>
```
Comment (PowerfulBacon) [inline tgui/packages/common/fp.js:13]: This format of separating the parameters of the functions is super ugly imo

File: tgui/packages/common/math.ts [hunk 2]
```diff
@@ -69,9 +69,7 @@ export const toFixed = (value, fractionDigits = 0) => {
  * Range is an array of two numbers, for example: [0, 15].
  */
 export const inRange = (value, range) => {
-  return range
-    && value >= range[0]
-    && value <= range[1];
+  return range && value >= range[0] && value <= range[1];
```
Comment (PowerfulBacon) [inline tgui/packages/common/math.ts:72]: It doesn't really work for things like this where it was better befire

File: tgui/packages/common/react.ts [hunk 3]
```diff
@@ -64,9 +64,7 @@ export const pureComponentHooks = {
  * A helper to determine whether the object is renderable by React.
  */
 export const canRender = (value: unknown) => {
-  return value !== undefined
-    && value !== null
-    && typeof value !== 'boolean';
+  return value !== undefined && value !== null && typeof value !== 'boolean';
```
Comment (PowerfulBacon) [inline tgui/packages/common/react.ts:67]: Here too

**Review by PowerfulBacon — COMMENTED** [review]
There's a lot of cases where it actually looks like a worse format after running through the formatter, especially with things that someone has split into multiple lines for clarity. Also, is there anything that actually enforces the style be maintained?
Also this will cause a huge number if conflicts and just be a pain in the ass in general

File: tgui/packages/common/math.ts (hunk 2 above)
Comment (itsmeow) [reply tgui/packages/common/math.ts:72]: This is the price you pay

File: tgui/packages/common/fp.js (hunk 1 above)
Comment (itsmeow) [reply tgui/packages/common/fp.js:13]: it's based on line lengths

**Comment by itsmeow** [comment]
It's maintained by the linter. I addressed the problem of conflicts.

**Comment by itsmeow** [comment]
The PR isn't "formats everything for max readability" it's "formats everything for max compatibility".

PrettierX creates the same result for the same input, making collaborative software 10x easier as any one change doesn't change the entire file's format if you have conflicting formatters.

**Review by PowerfulBacon — APPROVED** [review]

**Comment by EvilDragonfiend** [comment]
by the way, TG has prettier check linter.

**Review by Crossedfall — APPROVED** [review]

**Pull request merged**